exercises/chapter7/src/Data/AddressBook/Validation.purs

```diff
@@ -10,74 +10,129 @@ import Data.Traversable (traverse)
 import Data.Validation.Semigroup (V, unV, invalid)
 import Partial.Unsafe (unsafePartial)
 
+-----------------
+-- Some simple early examples returning `Either` instead of `V`:
+
+-- ANCHOR: nonEmpty1
+nonEmpty1 :: String -> Either String Unit
+nonEmpty1 "" = Left "Field cannot be empty"
+nonEmpty1 _  = Right unit
+-- ANCHOR_END: nonEmpty1
+
+-- ANCHOR: validatePerson1
+validatePerson1 :: Person -> Either String Person
+validatePerson1 p =
+  person <$> (nonEmpty1 p.firstName *> pure p.firstName)
+         <*> (nonEmpty1 p.lastName  *> pure p.lastName)
+         <*> pure p.homeAddress
+         <*> pure p.phones
+-- ANCHOR_END: validatePerson1
```

Comment:
Not sure about the best way to resolve this naming collision between this early version that returns `Either String Unit` and the later example that returns `V Errors Unit`.

Just adding a `1` suffix to this early unused version. The prime suffix is taken by `ValidatePerson'` at the end of this file.